SourceCgit/SourceCgit.php

```diff
@@ -250,12 +250,12 @@ public function commit_revision( $p_input ) {
 	 * @return array author / committer
 	 */
 	public function commit_author( $p_input ) {
-		preg_match( "#<tr><th>author</th><td>(.*?</span>|.*?)(.*?) <(.*?@.*?)></td><td class='right'>(.*?)</td>#", $p_input, $t_matches);
+		preg_match( "#<tr><th>author</th><td>(.*?</span>|.*?)(.*?) <((.*?@.*?)|)></td><td class='right'>(.*?)</td>#", $p_input, $t_matches);
```

Comment:
Unless I'm missing something, `(.*?</span>|.*?)` seems unnecessarily complex. You're just trying to ignore anything between the `<td>` and the closing `</span>` tags, right ? If so, `(.*?</span>)?` would be better IMO. And since the data is not being used, you should make that a non-capturing group: `(?:.*?</span>)?`.

The regex construct `((.*?@.*?)|)` seems a bit strange - why add an extra capturing group with an *or* ? I would think `(.*?@.*?)?` would be simpler and more efficient. 

Also, the extra group causes a shift in preg_match() output array, so the last group just before the `</td>` (the timestamp) will be number 5 instead of 4, so you'll have a problem with the timestamp (`$t_commit['date']` assignment)

Reply:
Regex are my weakness :,) thank you for that, I'ven't thought about when I've wrote it.

Reply:
About the shift in preg_match(), I don't have this issue. It's work just fine and show the date.